domain/secret/service/service.go

```diff
@@ -151,14 +52,17 @@ var NotImplementedBackendConfigGetter = func(context.Context) (*provider.ModelBa
 
 // SecretService provides the API for working with secrets.
 type SecretService struct {
-	st                State
-	logger            logger.Logger
-	clock             clock.Clock
-	providerGetter    func(backendType string) (provider.SecretBackendProvider, error)
-	adminConfigGetter BackendAdminConfigGetter
+	secretState                   State
+	secretBackendReferenceMutator SecretBackendReferenceMutator
+	logger                        logger.Logger
+	clock                         clock.Clock
+	providerGetter                func(backendType string) (provider.SecretBackendProvider, error)
+	adminConfigGetter             BackendAdminConfigGetter
 
 	activeBackendID string
+	modelID         coremodel.UUID
```

Comment:
services which operate on a model should not have a model uuid attribute.
The model uuid is looked up from the db.
The state struct already has that capability.